all_deck_summary_stats.py

```diff
@@ -0,0 +1,143 @@
+count = 0
+number_of_decks = len(os.listdir(directory))
+
+for filename in os.listdir(directory):
+    if filename.endswith(".json"): 
+        #print(os.path.join(directory, filename))
+
+        start_time = time.time()
+        
+        json_file = directory + filename
+        
+        with open(json_file) as data_file:
+            json_data = json.load(data_file)
+        
+        deck_number = list(json_data.keys())[0]
+        deck_object = json_data[deck_number]
+        list_of_card_groups = deck_object['list']
+        
+        card_type_dfs = map(get_df_of_cards_in_type, list_of_card_groups.keys())
+        deck_df = pd.concat(card_type_dfs)
+        deck_df['how_many_the_deck_needs'] = pd.to_numeric(deck_df['how_many_the_deck_needs'])
+        
+        #Skip if user didn't enter real cards and no price info available (e.g., https://www.mtggoldfish.com/deck/378046#paper):
+        if 'price' not in deck_df:
+            continue 
+        
+        deck_df['price'] = pd.to_numeric(deck_df['price'].str.replace(',','').fillna(0))
+        deck_df['unit_price'] = deck_df['price'] / deck_df['how_many_the_deck_needs']
+        
+        deck_df['how_many_you_have'] = deck_df.apply(get_how_many_you_have, axis=1)
+        deck_df['how_many_you_need'] = deck_df.apply(get_how_many_you_need, axis=1)
+        cards_you_have = deck_df[deck_df['how_many_you_have'] > 0]        
+        
+        deck_df['value_of_cards_you_have'] = deck_df.apply(get_value_of_cards_you_have, axis=1)
+        deck_df['cost_to_complete_deck_requirements'] = deck_df.apply(get_cost_to_complete_deck_requirements, axis=1)
+        
+        deck_name = deck_object['description']['deck_name']
+        archetype = deck_object['description']['archetype']
+        
+        cost_owned = deck_df['value_of_cards_you_have'].sum()
+        cost_total = deck_df['cost_to_complete_deck_requirements'].sum()
+        pct_cost_owned = division_by_zero_to_zero(cost_owned, cost_total)
+        
+        cards_owned = deck_df['how_many_the_deck_needs'].where(deck_df['how_many_you_have'] > 0, 0).sum()
+        cards_total = deck_df['how_many_the_deck_needs'].sum()
+        pct_cards_owned = division_by_zero_to_zero(cards_owned, cards_total)
+        
+        pokedex_owned = len(cards_you_have)
+        pokedex_total = len(deck_df)
+        pct_pokedex_owned = division_by_zero_to_zero(pokedex_owned, pokedex_total)
+        
+        summary = pd.DataFrame(data=[[
+            deck_name,
+            archetype,
+            deck_number,
+            pct_cost_owned,
+            pct_pokedex_owned,
+            pct_cards_owned,
+            cost_owned,
+            cost_total,
+            pokedex_owned,
+            pokedex_total,
+            cards_owned,
+            cards_total
+        ]],columns = output_columns)
+        
+        #output = output.append(summary)
+        output, summary = output.align(summary, axis=1)
+        output = pd.concat([output, summary])
```

Comment:
I'm not sure what the slow time increase is from, but doing high level operations for every file may be part of it. Could try just making a big list by mapping over the files, then creating a dataframe from the list, instead of basically doing a big 'reduce' here.